docs/style-guide.rst

```diff
@@ -1275,6 +1275,10 @@ Enums
 
 Enums, in the style of simple type declarations, should be named using the CapWords style. Examples: ``TokenGroup``, ``Frame``, ``HashStyle``, ``CharacterLocation``.
 
+Immutable variables
+===================
+
+
```

Comment:
@ekpyron and @cameel is it CONSTANTS_WITH_UNDERSCORES or camelCase?

Reply:
Good question :) Ether one has something going for it.

On one hand, I think that the choice between a state variable and an immutable is sometimes just an implementation detail. So a convention matching state variables would make sense. On the other hand, there are important semantic differences between the two and some people may want immutables to stand out precisely to make it clear which one you're dealing with.

I have no personal opinion on that matter.

Looks like [solhint does not have a style rule for immutables either](https://github.com/protofire/solhint/blob/master/docs/rules.md#style-guide-rules). Only for constants.

Looking at examples from our external tests, people seem to be using both, though the style matching state variables seems to be more common.

Reply:
In terms of consistency with other parts of Solidity, what would be your guess between one or the other?

Reply:
I wouldn't call either of them inconsistent. Both follow existing features and immutables have some features of both constants and state variables.

I'm not sure if making a choice about it is now is a good idea. I'd be inclined to just mention both possibilities and wait until someone complains and finds good arguments to choose one over the other.

Reply:
Thank you Kamil!
@ekpyron any opinion here?

Reply:
Not really :-). Probably best to just search github for the most common practice and just document that. It'd make sense to align it with constants or with state variables or to make it deliberately distinct from either of them.

Reply:
3 ORs. 😆 Cool, closing this.